march_joint_inertia_controller/include/march_joint_inertia_controller/joint_trajectory_inertia_controller.h

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020 Project March.
+
+#ifndef MARCH_HARDWARE_JOINT_TRAJECTORY_INERTIA_CONTROLLER_H
+#define MARCH_HARDWARE_JOINT_TRAJECTORY_INERTIA_CONTROLLER_H
+
+#include <cassert>
+#include <control_toolbox/pid.h>
+#include <controller_interface/controller.h>
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include <joint_trajectory_controller/joint_trajectory_controller.h>
+#include "march_joint_inertia_controller/inertia_estimator.h"
+#include <memory>
+#include <pluginlib/class_list_macros.h>
+#include <realtime_tools/realtime_buffer.h>
+#include <realtime_tools/realtime_publisher.h>
+#include <ros/node_handle.h>
+#include <ros/time.h>
+#include <trajectory_interface/quintic_spline_segment.h>
+
+namespace joint_trajectory_controller
+{
+typedef trajectory_interface::QuinticSplineSegment<double> SegmentImpl;
+typedef JointTrajectorySegment<SegmentImpl> Segment;
+typedef typename Segment::State State;
+}  // namespace joint_trajectory_controller
+
+template <>
+class HardwareInterfaceAdapter<hardware_interface::EffortJointInterface, joint_trajectory_controller::State>
```

Comment:
I don't fully understand what this class does and what the difference is between the `IntertiaController`. As far as I can tell, this "controller" does not implement the controller interface like `IntertiaController` does. And why does it use an empty template?

Reply:
The main difference between the controllers is that this one accepts trajectory commands, whereas the other needs a specified reference value meaning; this one is the same as the one used on the exo already. The other is a nice to have as far as I see it. I have put some work into fixing it before realizing it was not what we directly needed.